pandas/_libs/src/vendored/numpy/datetime/np_datetime.c

```diff
@@ -752,66 +752,57 @@ void pandas_timedelta_to_timedeltastruct(npy_timedelta td,
     }
 
     const npy_int64 per_day = sec_per_day * per_sec;
-    npy_int64 frac;
+    const int sign = td < 0 ? -1 : 1;
     // put frac in seconds
-    if (td < 0 && td % per_sec != 0)
-      frac = td / per_sec - 1;
-    else
-      frac = td / per_sec;
+    npy_int64 sfrac = td / per_sec;
+    if (sign < 0) {
+      if (td % per_sec != 0)
+        sfrac -= 1;
 
-    const int sign = frac < 0 ? -1 : 1;
-    if (frac < 0) {
       // even fraction
-      if ((-frac % sec_per_day) != 0) {
-        out->days = -frac / sec_per_day + 1;
-        frac += sec_per_day * out->days;
+      if ((-sfrac % sec_per_day) != 0) {
+        out->days = sfrac / sec_per_day - 1;
+        sfrac -= sec_per_day * out->days;
```

Comment:
Is this exactly the same as before? Previously it looks like out->days would be positive but this is now negative?

Sorry if misreading - again just want to be careful as I'm not sure how well our test cases are hitting all of these branches